src/factory/mod.rs

```diff
@@ -216,13 +287,32 @@ impl Factory {
                     .map(|t| [f2i(t.x), f2i(t.y), f2i(t.z), f2i(t.w)]),
             )
         };
-        izip!(position_iter, normal_iter, tangent_iter, uv_iter)
-            .map(|(position, normal, tangent, tex_coord)| {
+        let joints_iter = if shape.joints.is_empty() {
+            Either::Left(iter::repeat([0.0, 0.0, 0.0, 0.0]))
+        } else {
+            Either::Right(shape.joints.iter().cloned())
+        };
+        let weights_iter = if shape.weights.is_empty() {
+            Either::Left(iter::repeat([1.0, 0.0, 0.0, 0.0]))
```

Comment:
N.B!

Reply:
This isn't as special as I originally thought. The new behaviour is to take the weights `[1.0, 1.0, 1.0, 1.0]` with joints `[0.0, 0.0, 0.0, 0.0]` so that an identity matrix is computed when there is no skeleton (i.e. the default joint transform buffer is used).